packages/snapshot-listener/src/index.ts

```diff
@@ -73,42 +63,46 @@ registerRoute(app, 'post', '/snapshot', async (req: Request, res: Response) => {
       res.status(400).send('Error sending snapshot event');
     }
 
-    const response = await fetchNewSnapshotProposal(parsedId, eventType);
+    const response = await fetchNewSnapshotProposal(parsedId);
+
+    const space = response.data.proposal?.space.id;
+
+    if (!space) {
+      log.error('Space not defined!');
+      return res.status(400).send('Error getting snapshot space');
+    }
+
+    const associatedCommunities = await models.CommunitySnapshotSpaces.count({
+      where: { snapshot_space_id: space },
+    });
+
+    if (associatedCommunities === 0) {
+      log.info(`No associated communities found for space ${space}`);
+      return res.status(200).json({ message: 'No associated community' });
+    }
 
-    const event: EventContext<'SnapshotProposalCreated'> = {
-      name: 'SnapshotProposalCreated',
-      payload: {
+    await models.Outbox.create({
```

Comment:
emitting to outbox inside a transaction will be a very common pattern for us, worth a high level abstraction (hooks or whatever is simpler)

Reply:
Yea will be using hooks. ChainEvents and Snapshots were the 2 exceptions since they don't have a related table in the DB. Note: we technically do have a `SnapshotProposal` table but it is almost completely useless and I intend to remove it very soon (hopefully clean-up Friday) https://github.com/hicommonwealth/commonwealth/issues/5252.

Reply:
To as well, if I create a utility it will be part of https://github.com/hicommonwealth/commonwealth/issues/6939